doozerlib/distgit.py

```diff
@@ -2000,43 +2021,54 @@ def extract_version_release_private_fix(self) -> (str, str, str):
             return version, prev_release, private_fix
         return None, None, None
 
-    def rebase_dir(self, version, release):
-        dg_path = self.dg_path
-        df_path = dg_path.joinpath('Dockerfile')
-        prev_release = None
-        with Dir(self.distgit_dir):
-            prev_version, prev_release, prev_private_fix = self.extract_version_release_private_fix()
-            if version is None and not self.runtime.local:
-                # Extract the previous version and use that
-                version = prev_version
-
-            # Make our metadata directory if it does not exist
-            util.mkdirs(dg_path.joinpath('.oit'))
-
-            # If content.source is defined, pull in content from local source directory
-            if self.has_source():
-                self._merge_source()
-
-                # before mods, check if upstream source version should be used
-                # this will override the version fetch above
-                if self.metadata.config.get('use_source_version', False):
-                    dfp = DockerfileParser(str(df_path))
-                    version = dfp.labels["version"]
-            else:
-                self.private_fix = bool(prev_private_fix)  # preserve private_fix boolean for distgit-only repo
-
-            # Source or not, we should find a Dockerfile in the root at this point or something is wrong
-            assertion.isfile(df_path, "Unable to find Dockerfile in distgit root")
+    def rebase_dir(self, version, release, terminate_event):
+        try:
+            # If this image is FROM another group member, we need to wait on that group member to determine if there are embargoes in that group member.
+            image_from = Model(self.config.get('from', None))
+            if image_from.member is not Missing:
+                self.wait_for_rebase(image_from.member, terminate_event)
```

Comment:
there is one member (`ose-metering-ansible-operator`) that uses other members as build stages (not just the final stage). i think this should loop through the `from` and the `builder` and wait on any members (and be private if any of them are private). i _think_ the ordering takes build stages into account. something like:

```suggestion
            parent_images = [Model(self.config.get('from', None))]
            parent_images.extend(self.config.get('from', {}).get('builder', [])))
            for image_from in parent_images:
                if image_from.member is not Missing:
                    self.wait_for_rebase(image_from.member, terminate_event)
```

Actually, that's not too pretty, and we'd have to do something similar at the part above determining if any of them are private... determining parents should be abstracted out.

Reply:
Yes, we are missing that part. I'll create a ticket in the backlog.

Reply:
https://issues.redhat.com/browse/ART-2349